runtime/src/main/java/org/corfudb/runtime/collections/streaming/DeltaStream.java

```diff
@@ -0,0 +1,184 @@
+package org.corfudb.runtime.collections.streaming;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.Buffer;
+import org.apache.commons.collections.BufferUtils;
+import org.apache.commons.collections.buffer.BoundedFifoBuffer;
+import org.corfudb.common.metrics.micrometer.MicroMeterUtils;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.runtime.exceptions.TrimmedException;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.runtime.view.AddressSpaceView;
+import org.corfudb.runtime.view.ReadOptions;
+import org.corfudb.runtime.view.stream.StreamAddressSpace;
+
+import java.time.Duration;
+import java.util.UUID;
+
+/**
+ * Unlike the {@link org.corfudb.runtime.view.stream.IStreamView} this stream implementation only tracks updates
+ * that haven't been checkpointed.
+ */
+@Slf4j
+public class DeltaStream {
+
+    private final AddressSpaceView addressSpaceView;
+
+    @Getter
+    private final UUID streamId;
+
+    /**
+     * Last Address consumed/read by this stream
+     */
+    private volatile long lastAddressRead;
+
+    /**
+     * Max seen address for this stream from the last refresh/sync
+     */
+    @Getter
+    private volatile long maxAddressSeen;
+
+    /**
+     * The last trim mark that has been observed
+     */
+    private volatile long trimMark;
+
+    /**
+     * The size of this stream's buffer (i.e., pending addresses to read)
+     */
+    private final int bufferSize;
+
+    /**
+     * A circular buffer that holds the queued address to read
+     */
+    private final Buffer addressesToRead;
+
+    private final ReadOptions options = ReadOptions
+            .builder()
+            .clientCacheable(false)
+            .ignoreTrim(false)
+            .waitForHole(true)
+            .serverCacheable(false)
+            .build();
+
+    public DeltaStream(AddressSpaceView addressSpaceView, UUID streamId, long lastAddressRead, int bufferSize) {
```

Comment:
One suggestion, I think besides the streamId, adding the streamTag would ease debugging (have it for logging purposes). We typically want to track updates to a given tag, and its easier too grep based on the name which is what the user knows about.

Reply:
Not sure what you mean. Add it to each log? or added on creation ?